src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/JavaScriptEncoder.cs

```diff
@@ -119,6 +119,7 @@ public unsafe override int FindFirstCharacterToEncode(char* text, int textLength
             {
                 throw new ArgumentNullException(nameof(text));
```

Comment:
@steveharter I meant it this way:
```c#
        [MethodImpl(MethodImplOptions.AggressiveInlining)]
        public unsafe override int FindFirstCharacterToEncode(char* text, int textLength)
        {
            if (text == null)
            {
                ThrowHelper_ArgumentNullException();
            }

            return _allowedCharacters.FindFirstCharacterToEncode(text, textLength);

            void ThrowHelper_ArgumentNullException()
            {
                throw new ArgumentNullException(nameof(text));
            }
        }
```

Reply:
I just reverted back to the original; I don't think having the ThrowHelper helps here with perf.